client/lib/media/actions.js

```diff
@@ -213,23 +219,37 @@ MediaActions.edit = function( siteId, item ) {
 };
 
 MediaActions.update = function( siteId, item ) {
-	var newItem;
-
 	if ( Array.isArray( item ) ) {
 		item.forEach( MediaActions.update.bind( null, siteId ) );
 		return;
 	}
 
-	newItem = assign( {}, MediaStore.get( siteId, item.ID ), item );
+	const mediaId = item.ID;
+	const newItem = assign( {}, MediaStore.get( siteId, item.ID ), item );
 
-	Dispatcher.handleViewAction( {
+	// let's updated data immediately
+	const updateAction = {
 		type: 'RECEIVE_MEDIA_ITEM',
-		siteId: siteId,
+		siteId,
 		data: newItem
-	} );
+	};
+
+	if ( item.media ) {
+		// also print edited image as soon as possible
```

Comment:
Maybe update to comment to be more descriptive? Like:

`Show a fake transient media item that can be rendered into the list immediately, even before the media has persisted to the server`